packages/ui/src/controlled-modal/index.tsx

```diff
@@ -96,7 +96,7 @@ export const ControlledModal: MemoExoticComponentWithAs<ModalProps, 'dialog'> =
     const { children, isOpen = false, handleClose, noAnimation = false, ...rest } = props;
     const onCloseHandler = isOpen ? handleClose : () => null;
 
-    useOnClickOutside(_ref, isOpen ? onCloseHandler : null);
+    useOnClickOutside(_ref, onCloseHandler);
```

Comment:
Can you go into more detail about why this needs to be the case? I made this change originally because having the handler always on is a significant performance drain.